clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetchMetricsManagerTest.java

```diff
@@ -66,14 +69,41 @@ public void tearDown() {
 
     @Test
     public void testLatency() {
-        metricsManager.recordLatency(123);
+        metricsManager.recordLatency("", 123);
         time.sleep(metrics.config().timeWindowMs() + 1);
-        metricsManager.recordLatency(456);
+        metricsManager.recordLatency("", 456);
 
         assertEquals(289.5, metricValue(metricsRegistry.fetchLatencyAvg), EPSILON);
         assertEquals(456, metricValue(metricsRegistry.fetchLatencyMax), EPSILON);
     }
 
+    @Test
+    public void testNodeLatency() {
+        String connectionId = "0";
+        MetricName nodeLatencyAvg = metrics.metricName("request-latency-avg", "group");
+        MetricName nodeLatencyMax = metrics.metricName("request-latency-max", "group");
+        registerNodeLatencyMetric(connectionId, nodeLatencyAvg, nodeLatencyMax);
+
+        metricsManager.recordLatency(connectionId, 123);
+        time.sleep(metrics.config().timeWindowMs() + 1);
+        metricsManager.recordLatency(connectionId, 456);
+
+        assertEquals(289.5, metricValue(metricsRegistry.fetchLatencyAvg), EPSILON);
+        assertEquals(456, metricValue(metricsRegistry.fetchLatencyMax), EPSILON);
+
+        assertEquals(289.5, metricValue(nodeLatencyAvg), EPSILON);
+        assertEquals(456, metricValue(nodeLatencyMax), EPSILON);
+
+        // Record metric against another node.
+        metricsManager.recordLatency("1", 501);
+
+        assertEquals(360, metricValue(metricsRegistry.fetchLatencyAvg), EPSILON);
+        assertEquals(501, metricValue(metricsRegistry.fetchLatencyMax), EPSILON);
+        // Node specific metric should not be affected.
+        assertEquals(289.5, metricValue(nodeLatencyAvg), EPSILON);
+        assertEquals(456, metricValue(nodeLatencyMax), EPSILON);
```

Comment:
Could we further verify the metric value for node 1?

Reply:
As the node metrics get registered in Selector.java, which cannot be spawned from FetchMetricsManagerTest, hence I added a node latency metric in tests to verify if the value is recorded for the node.

As in the tests, we are not registering any node 1 sensor/metric hence it won't be found, so I don't find any valid validation here. I did check if the recordLatency method is called for some other node, then node-0 metrics should be unchanged.